### PR TITLE
chore(ty): clean up type-check config and replace dead type ignores

### PR DESCRIPTION
**Pull request by Alberto-Codes**
The ty type-check CI gate carried stale overrides from ty 0.0.1a33 and 13 `# type: ignore` comments using mypy syntax instead of the project-standard `# ty: ignore[rule]` syntax. This cleans up the config, replaces the comments with proper ty-native suppressions or type-narrowing fixes, and updates project guidance.

- Remove 3 stale ty test overrides (`missing-argument`, `possibly-missing-attribute`, `unused-ignore-comment`) now that ty 0.0.18 resolves them
- Replace 13 `# type: ignore` comments in src/: 6 fixed with type-narrowing asserts, 7 converted to targeted `# ty: ignore[rule]`
- Update project-context.md Type Checking section with ty syntax guidance and fix-first philosophy
- Add 29 unit tests for `_aggregate_acceptance_score()` and `_extract_json_from_text()` covering error paths, edge cases, and fallback chains

Test: `uv run pytest -v && uv run ty check src tests`

test(scoring): add unit tests for score aggregation and JSON extraction

- 17 tests for _aggregate_acceptance_score: empty list, NaN/Inf/-Inf, mixed non-finite, sum vs mean metrics, negative scores, zero-score edge cases
- 12 tests for _extract_json_from_text: direct parse, markdown code blocks, brace-matching fallback, nested objects, invalid content, empty string

---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass (`uv run pytest`)
- [x] Lint passes (`uv run ruff check .`)
- [ ] Breaking changes use `!` in title and `BREAKING CHANGE:` in body

### Review Focus
- Verify the 3 removed ty overrides are genuinely stale (no new diagnostics)
- Check that type-narrowing asserts in adk_adapter.py and multi_agent.py are safe runtime guards
- Confirm new `# ty: ignore[rule]` comments use correct rule names

### Related
- Story 1B.5 (created in epics.md) tracks elimination of remaining ty:ignore comments through typed Result patterns

### CODE REVIEW

File: tests/unit/engine/test_aggregate_acceptance_score.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+pytestmark = pytest.mark.unit
+
+
+@pytest.fixture
+def make_engine():
+    """Factory to create an engine with a given acceptance_metric."""
+
+    def _factory(metric: str = "mean") -> AsyncGEPAEngine:
+        adapter = create_mock_adapter()
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply tests/unit/engine/test_aggregate_acceptance_score.py]: Fixed in d549232 — added `-> Callable[..., AsyncGEPAEngine]` return type annotation.

File: src/gepa_adk/adapters/evolution/multi_agent.py [hunk 2]
```diff
@@ -679,6 +679,8 @@ async def evaluate(
 
         Returns:
             EvaluationBatch containing outputs, scores, and optional trajectories.
+                Gather results are type-narrowed with runtime assertions for
+                ty type-checker compatibility.
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply src/gepa_adk/adapters/evolution/multi_agent.py]: Fixed in d549232 — same indentation fix applied.
